tokenizers/src/icu.rs

```diff
@@ -43,9 +44,14 @@ impl<'a> From<&'a str> for ICUBreakingWord<'a> {
     fn from(text: &'a str) -> Self {
         let loc = rust_icu_uloc::get_default();
         let ustr = &UChar::try_from(text).expect("is an encodable character");
+        // Implementation from a similar fix in https://github.com/jiegec/tantivy-jieba/pull/5
+        // referenced by Tantivy issue https://github.com/quickwit-oss/tantivy/issues/1134
+        let mut char_indices = text.char_indices().collect::<Vec<_>>();
+        char_indices.push((text.len(), '\0'));
```

Comment:
What's the reason for adding a null byte at the end of the char_indices list?

Reply:
Was getting an index out of bounds error without it. I lifted it from the implementation linked in the comments.

Reply:
Can you make a standalone test which demonstrates the oob panic?

Reply:
All of the current icu unit tests would fail if this line were removed. The reason is because Tantivy defines the end of a Token to be "offset of the last byte + 1". https://docs.rs/tantivy/latest/tantivy/tokenizer/struct.Token.html. This is equivalent to "offset of the first byte of the next token." If we don't add the null byte, then getting the end of the last token will fail.

Reply:
done!